sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/TimeUtilsTest.groovy

```diff
@@ -0,0 +1,126 @@
+package com.azure.storage.blob.changefeed
+
+import com.azure.storage.blob.changefeed.implementation.util.TimeUtils
+import spock.lang.Specification
+
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+
+class TimeUtilsTest extends Specification {
+
+    def "convertPathToTime"() {
+        expect:
+        TimeUtils.convertPathToTime(path) == time
+
+        where:
+        path                                     || time
+        null                                     || null
+        "idx/segments/2019"                      || OffsetDateTime.of(2019, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/"                     || OffsetDateTime.of(2019, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11"                   || OffsetDateTime.of(2019, 11, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11/"                  || OffsetDateTime.of(2019, 11, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11/02"                || OffsetDateTime.of(2019, 11, 2, 0, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11/02/"               || OffsetDateTime.of(2019, 11, 2, 0, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11/02/1700"           || OffsetDateTime.of(2019, 11, 2, 17, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11/02/1700/"          || OffsetDateTime.of(2019, 11, 2, 17, 0, 0, 0, ZoneOffset.UTC)
+        "idx/segments/2019/11/02/1700/meta.json" || OffsetDateTime.of(2019, 11, 2, 17, 0, 0, 0, ZoneOffset.UTC)
```

Comment:
Is there any need to test a path like `idx/segments/2019/11/02/1730/meta.json` at all?

Reply:
No real need but I was just including all possible paths in Changefeed Segments

Reply:
Sorry I think the intention of my question was unclear. I was highlighting 1730 as opposed to 1700. Basically something that doesn't align along an hour.

Reply:
I don't think we ever expect it though just cause the segments are hourly